src/angular-cesium-widgets/services/zoom-to-rectangle.service.ts

```diff
@@ -0,0 +1,243 @@
+ *  backgroundColor - optional - the background color of the rectangle element - default: 'transparent'
+ *  resetKeyCode - optional - the key code of the key that is used to reset the drawing of the rectangle - default: 27 (ESC key)
+ * }
+ * @param mapId - optional - the mapId of the map that the tool will be used in.
+ *
+ */
+
+@Injectable()
+export class ZoomToRectangleService {
+  private cameraService: CameraService;
+  private cesiumService: CesiumService;
+  constructor(
+    private mapsManager: MapsManagerService,
+    @Optional() cameraService: CameraService,
+    @Optional() cesiumService: CesiumService,
+  ) {}
+
+  init(cesiumService: CesiumService, cameraService: CameraService) {
+    this.cameraService = cameraService;
+    this.cesiumService = cesiumService;
+  }
+
+  private mapsZoomElements = new Map<string, ZoomData>();
+  private defaultOptions = {
+    animationDurationInSeconds: 0.5,
+    resetKeyCode: 27,
+    borderStyle: '2px solid rgba(0,0,0,0.5)',
+    backgroundColor: 'rgba(0,0,0,0.2)',
+    autoDisableOnZoom: true,
+  };
+
+  activate(
+    options: {
+      onStart?: (acMap?: AcMapComponent) => any;
+      onComplete?: (acMap?: AcMapComponent) => any;
+      autoDisableOnZoom?: boolean;
+      animationDurationInSeconds?: number;
+      borderStyle?: string;
+      backgroundColor?: string;
+      resetKeyCode?: number;
+    } = {},
+    mapId?: string,
+  ) {
+    if ((!this.cameraService || !this.cesiumService) && !mapId) {
+      throw new Error(`The function must receive a mapId if the service wasn't initialized`);
+    }
+    const finalOptions = Object.assign({}, this.defaultOptions, options);
+    let cameraService = this.cameraService;
+    let mapContainer;
+    let map;
+    if (this.cesiumService) {
+      mapContainer = this.cesiumService.getViewer().container;
+      map = this.cesiumService.getMap();
+    }
+    if (mapId) {
+      map = this.mapsManager.getMap(mapId);
+      if (!map) {
+        throw new Error(`Map not found with id: ${mapId}`);
+      }
+      cameraService = map.getCameraService();
+      mapContainer = map.getCesiumViewer().container;
+    }
+
+    if (!cameraService || !mapContainer) {
+      throw new Error(`The function must receive a mapId if the service wasn't initialized`);
+    }
+    this.disable(mapId);
+    const container = document.createElement('div');
+    mapContainer.style.position = 'relative';
+    container.style.position = 'absolute';
+    container.style.width = '100%';
+    container.style.height = '100%';
+    container.style.top = '0';
+    container.style.left = '0';
+    mapContainer.appendChild(container);
+    const mapZoomData: ZoomData = { container };
+    this.mapsZoomElements.set(mapId || this.cesiumService.getMap().getId(), mapZoomData);
+    let mouse = {
+      endX: 0,
+      endY: 0,
+      startX: 0,
+      startY: 0,
+    };
+    let borderElement: HTMLElement | undefined;
+
+    container.onmousedown = e => {
```

Comment:
shouldnt the event be configurable? like if i want zoom with right click.

Reply:
I think we can leave it like that for now. If we decide it needs modefing will do it in a new PR